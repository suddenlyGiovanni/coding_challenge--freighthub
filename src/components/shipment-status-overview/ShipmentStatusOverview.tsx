import * as React from 'react'
import styled from '@emotion/styled/macro'
import LinearProgress from '@material-ui/core/LinearProgress'

import { Status } from 'features/shipments/typings'
import { Location } from './Location'

const OverviewContainer = styled.div`
  position: relative;

  display: flex;
  align-items: center;
  flex-direction: row;
  justify-content: center;

  width: 100%;
  height: 100%;
`

const Temp = styled.div`
  position: relative;

  display: block;
  flex: 1;
`

interface Props {
  status: Status
  origin: string
  destination: string
}
export const ShipmentStatusOverview: React.FC<Props> = ({
  status,
  origin,
  destination,
}) => {
  return (
    <OverviewContainer>
      <Location name={origin} active={true} alignment="left" />
      <Temp>
        <LinearProgress
          variant="determinate"
          value={status === 'COMPLETED' ? 100 : 50}
          style={{
            margin: '0 10px',
          }}
        />
      </Temp>
      <Location
        name={destination}
        active={status === 'COMPLETED'}
        alignment="right"
      />
    </OverviewContainer>
  )
}
