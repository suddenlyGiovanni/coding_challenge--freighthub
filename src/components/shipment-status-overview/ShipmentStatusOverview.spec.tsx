import * as React from 'react'
import { render } from '@testing-library/react'

import { ShipmentStatusOverview, Props } from './ShipmentStatusOverview'

describe('ShipmentStatusOverview', () => {
  test('snapshot', () => {
    // Arrange
    const props: Props = {
      status: 'ACTIVE',
      destination: 'Saarbr√ºcker Str. 38, 10405 Berlin',
      origin: 'Shanghai Port',
    }
    const { container } = render(<ShipmentStatusOverview {...props} />)

    // Assert
    expect(container).toMatchSnapshot()
  })
})
