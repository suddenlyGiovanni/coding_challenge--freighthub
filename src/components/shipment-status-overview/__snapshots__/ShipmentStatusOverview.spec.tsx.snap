// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShipmentStatusOverview snapshot 1`] = `
<div>
  <div
    class="css-1d6dqeu-OverviewContainer ecukxon0"
  >
    <div
      class="css-178mkk4-LocationContainer e17eqbe40"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15 11V5l-3-3-3 3v2H3v14h18V11h-6zm-8 8H5v-2h2v2zm0-4H5v-2h2v2zm0-4H5V9h2v2zm6 8h-2v-2h2v2zm0-4h-2v-2h2v2zm0-4h-2V9h2v2zm0-4h-2V5h2v2zm6 12h-2v-2h2v2zm0-4h-2v-2h2v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <p
        class="MuiTypography-root css-i8g5x1-PlaceName e17eqbe41 MuiTypography-body2 MuiTypography-noWrap"
        data-testid="place-name"
        style="left: 0px;"
      >
        Shanghai Port
      </p>
    </div>
    <div
      class="css-ukvtdo-Temp ecukxon1"
    >
      <div
        aria-valuenow="50"
        class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
        role="progressbar"
        style="margin: 0px 10px;"
      >
        <div
          class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
          style="transform: translateX(-50%);"
        />
      </div>
    </div>
    <div
      class="css-178mkk4-LocationContainer e17eqbe40"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M15 11V5l-3-3-3 3v2H3v14h18V11h-6zm-8 8H5v-2h2v2zm0-4H5v-2h2v2zm0-4H5V9h2v2zm6 8h-2v-2h2v2zm0-4h-2v-2h2v2zm0-4h-2V9h2v2zm0-4h-2V5h2v2zm6 12h-2v-2h2v2zm0-4h-2v-2h2v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <p
        class="MuiTypography-root css-i8g5x1-PlaceName e17eqbe41 MuiTypography-body2 MuiTypography-noWrap"
        data-testid="place-name"
        style="right: 0px;"
      >
        Saarbr√ºcker Str. 38, 10405 Berlin
      </p>
    </div>
  </div>
</div>
`;
